### ISSUE TITLE
[Feature Request] Increase streaming resolution/framerate

### ISSUE THREAD

**Issue by xxated** [jschmid1/gopro_as_webcam_on_linux#61]
Hello! Thanks for supporting this project!

Currently, the max setting for streaming is 1080p/30fps. 
I'm wondering if it would be possible to add some more configurations for resolution (4K/5.3K), and framerate (60fps)? 
Maybe also HDR, and color bitrate.
That would be very helpful !!!

Thanks!

**Comment by DavidWoli** [comment]
Is that supported from the GoPro itself? If not, than we have to change the firmware first (that is not easy at all).

**Comment by xxated** [comment]
> Is that supported from the GoPro itself? If not, than we have to change the firmware first (that is not easy at all).

Yes, it is supported. I have last GoPro, which is 12 Black. It supports filming in the 5.3K + HDR or 4K + 60fps.

**Comment by xxated** [comment]
> Is that supported from the GoPro itself? If not, than we have to change the firmware first (that is not easy at all).

Yes, it is supported. I have latest GoPro 12 Black, which supports filming in 5.3K + HDR in 30fps or 4K in 60/120fps. 

Sorry for accidentally closing the issue, mis-clicked from phone 😅

**Comment by DavidWoli** [comment]
@xxated I am not talking about the recording feature but the webcam streaming feature. Does GoPro itself support for example 4K webcam streaming. Technically it should work because the GoPro is able to record in that resolution but if the firmware does not send a 4K stream over the USB C cable, you cannot do anything (unless someone is willing to patch the firmware).

**Comment by xxated** [comment]
> @xxated I am not talking about the recording feature but the webcam streaming feature. Does GoPro itself support for example 4K webcam streaming. Technically it should work because the GoPro is able to record in that resolution but if the firmware does not send a 4K stream over the USB C cable, you cannot do anything (unless someone is willing to patch the firmware).

Yes, of course it supports streaming as well. I was able to stream from my camera in 5.3k to my phone from the GoPro Quik app.
